app/assets/javascripts/i18n.js

```diff
@@ -857,12 +854,43 @@
     }
 
     return scope;
-  }
+  };
+  /**
+   * Merge obj1 with obj2
+   * @param {Object} obj1 Default settings
+   * @param {Object} obj2 User obj2
+   * @returns {Object} Merged values of obj1 and obj2
+   */
+  I18n.extend = function ( obj1, obj2 ) {
+    var extended = {};
+    var prop;
+    for (prop in obj1) {
+      if (Object.prototype.hasOwnProperty.call(obj1, prop)) {
```

Comment:
The extend method was meant to be as simple as possible. I used the KISS method for it... you could  copy jQuery, Underscore, etc... extend method if you chose to but they are much more complex and could lead to instability.